clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -1678,6 +1680,62 @@ public void testMultipleAddPartitionsPerForOneProduce() throws InterruptedExcept
         assertTrue(secondResponseFuture.isDone());
     }
 
+    @ParameterizedTest
+    @ValueSource(ints = {3, 7, 14, 51})
```

Comment:
I think another way to do this is like this:
```java
@EnumSource(names = { 
  "UNKNOWN_TOPIC_OR_PARTITION", 
  "REQUEST_TIMED_OUT", 
  "COORDINATOR_LOAD_IN_PROGRESS", 
  "CONCURRENT_TRANSACTIONS" 
})
```

Reply:
I thought about this, but I didn't know how to make the errors from strings.